### ISSUE TITLE
gem install error "Permission denied" for jruby 9.3.2.0 [darwin-aarch64] (Apple M1)

### ISSUE THREAD

**Issue by dlobsinger** [jruby/jruby#6977]
**Environment Information**

- JRuby: jruby 9.3.2.0 (2.6.8) 2021-12-01 0b8223f905 OpenJDK 64-Bit Server VM 17.0.1+1 on 17.0.1+1 +jit [darwin-aarch64]
- OS: macOS Monterey v12.1 (`Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:41 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T6000 arm64`)
- Java: 
openjdk version "1.8.0_312"
OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-macos-aarch64) (build 1.8.0_312-b07)
OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-macos-aarch64) (build 25.312-b07, mixed mode)

**Behavior**
With a fresh install of JRuby v9.3.2.0 installing gems fails with an error 
```
ERROR:  While executing gem ... (Errno::EACCES)
    Permission denied - /opt/homebrew/Cellar/jruby/9.3.2.0/libexec/lib/ruby/gems/shared/cache/multi_xml-0.6.0.gem
```
Looking at the file permissions it seems to have extremely limited access 
```
> stat /opt/homebrew/Cellar/jruby/9.3.2.0/libexec/lib/ruby/gems/shared/cache/multi_xml-0.6.0.gem
16777233 3213596 ---------x 1 dave admin 0 15360 "Dec 20 14:28:33 2021" "Dec 20 14:28:33 2021" "Dec 20 14:28:33 2021" "Dec 20 14:28:33 2021" 4096 32 0 /opt/homebrew/Cellar/jruby/9.3.2.0/libexec/lib/ruby/gems/shared/cache/multi_xml-0.6.0.gem
```
This can be reproduced with the following operations
```
> brew install jruby 
> jruby -S gem install httparty (or any other)
```

This also affects the most recent changes on https://github.com/jruby/jruby/tree/jruby-9.2 related to JNR deps.

**Comment by 545ch4** [comment]
Same here.

When running with `gem install -V -f --backtrace` you'll see at `mkdir_p_safe` in `/libexec/lib/ruby/stdlib/rubygems/package.rb` that ech part of the path should be a directory. When using `jruby` on a apple-silicon this is not always the case... To reproduce:

```
$ mkdir /tmp/test
$ RBENV_VERSION=2.7.5 ruby -e "puts File.lstat('/tmp').inspect; puts File.lstat('/tmp').directory?; puts File.lstat('/tmp/test').inspect; puts File.lstat('/tmp/test').directory?"
#<File::Stat dev=0x1000011, ino=1152921500312809861, mode=0120755, nlink=1, uid=0, gid=0, rdev=0x0, size=11, blksize=4096, blocks=0, atime=2021-12-08 00:39:27 +0100, mtime=2021-12-08 00:39:27 +0100, ctime=2021-12-08 00:39:27 +0100, birthtime=2021-12-08 00:39:27 +0100>
false
#<File::Stat dev=0x1000011, ino=3196862, mode=040755, nlink=2, uid=501, gid=0, rdev=0x0, size=64, blksize=4096, blocks=0, atime=2021-12-28 14:25:01.768275781 +0100, mtime=2021-12-28 14:25:01.768275781 +0100, ctime=2021-12-28 14:25:01.768275781 +0100, birthtime=2021-12-28 14:25:01.768275781 +0100>
true
$ jruby -e "puts File.lstat('/tmp').inspect; puts File.lstat('/tmp').directory?; puts File.lstat('/tmp/test').inspect; puts File.lstat('/tmp/test').directory?"
#<File::Stat dev=0x1000011, ino=106989, mode=030605, nlink=14, uid=268435455, gid=0, rdev=0x0, size=0, blksize=0, blocks=1638920367, atime=1970-01-01 01:00:01 +0100, mtime=1970-01-01 01:00:01 +0100, ctime=1970-01-01 01:00:01 +0100, birthtime=2021-12-08 00:39:27 +0100>
false
#<File::Stat dev=0x1000011, ino=147949, mode=0143676, nlink=48, uid=0, gid=501, rdev=0x0, size=768275781, blksize=768275781, blocks=1640697901, atime=1970-01-01 01:00:01 +0100, mtime=1994-05-07 03:56:22 +0200, ctime=1994-05-07 03:56:22 +0200, birthtime=2021-12-28 14:25:01 +0100>
false
```

All jruby stat dates are looking weired, `uid` and `gid` are incorrect, and certainly `/tmp/test` **is** a directory.

Could be related/equal to https://github.com/jruby/jruby/issues/5521 => https://github.com/jnr/jnr-posix/issues/126

**Comment by headius** [comment]
Likely fixed by jnr/jnr-posix#173 which has merged into all branches. Marking as fixed in 9.3, which is the release that will officially support Apple Silicon.

See also #6985 and #6986.
